app/src/main/java/it/quip/android/network/FacebookClient.java

```diff
@@ -0,0 +1,63 @@
+package it.quip.android.network;
+
+
+import com.facebook.AccessToken;
+import com.facebook.GraphRequest;
+import com.facebook.GraphResponse;
+import com.facebook.HttpMethod;
+import com.facebook.login.LoginResult;
+
+
+public class FacebookClient {
+
+    public interface OnFacebookResponse {
+
+        void onFacebookResponse(GraphResponse response);
+
+    }
+
+    public static class FacebookClientException extends Exception {
+
+        public FacebookClientException(String detailedMessage) {
+            super(detailedMessage);
+        }
+
+    }
+
+    private static FacebookClient client;
+
+    private AccessToken accessToken;
+
+    public static FacebookClient newInstance(LoginResult loginResult) {
+        client = new FacebookClient(loginResult.getAccessToken());
+        return client;
+    }
+
+    public static FacebookClient getInstance() throws FacebookClientException {
+        if (client == null) {
+            throw new FacebookClientException("thangs");
```

Comment:
It should be throwing an exception. If this case ever occurs, something terribly wrong has happened. We can catch it at the top level and address it as needed.